lib/fog/libvirt/models/compute/server.rb

```diff
@@ -281,10 +283,28 @@ def to_xml
               end
 
               xml.vcpu(cpus)
-              xml.os do
+
+              os_tags = {}
+
+              # Secure boot and stateless UEFI both imply an EFI firmware
+              if ["secure", "stateless"].include?(os_loader)
+                os_tags[:firmware] = "efi"
+              elsif os_firmware
+                os_tags[:firmware] = os_firmware
+              end
+
+              xml.os(**os_tags) do
                 type = xml.type(os_type, :arch => arch)
                 type[:machine] = "q35" if ["i686", "x86_64"].include?(arch)
 
+                # TODO: can you use both secure and stateless at the same time?
+                case attributes[:os_loader]
```

Comment:
Note to self: https://libvirt.org/kbase/secureboot.html says the os_loader was only needed for libvirt >= 7.2 < 8.6